src/frontend/src/test-e2e/pinAuth.test.ts

```diff
@@ -39,6 +58,36 @@ test("Register and Log in with PIN identity", async () => {
   }, APPLE_USER_AGENT);
 }, 300_000);
 
+test("Register and log in with PIN identity, retry on wrong PIN", async () => {
+  await runInBrowser(async (browser: WebdriverIO.Browser) => {
+    const pin = "123456";
+
+    await browser.url(II_URL);
+    const userNumber = await FLOWS.registerPinWelcomeView(browser, pin);
+    const mainView = new MainView(browser);
+    await mainView.waitForDisplay(); // we should be logged in
+    await mainView.logout();
+
+    const welcomeView = new WelcomeView(browser);
+    await welcomeView.waitForDisplay();
+    await welcomeView.login();
+    await welcomeView.typeUserNumber(userNumber);
+    await browser.$("button[data-action='continue']").click();
+    const pinAuthView = new PinAuthView(browser);
+    await pinAuthView.waitForDisplay();
+    await pinAuthView.enterPin("456321"); // wrong PIN
```

Comment:
I would create a `wrongPin` variable next to the `pin` variable. This way you don't need the comment.